build/webpack.conf.js

```diff
@@ -1,65 +1,176 @@
+        })
+      },
+      {
+        // Match woff2 in addition to patterns like .woff?v=1.1.1.
+        test: /\.(woff|woff2)(\?v=\d+\.\d+\.\d+)?$/,
+        loader: 'url-loader',
         options: {
-          presets: ['env'],
-        },
-      }, {
+          // Limit at 50k. Above that it emits separate files
+          limit: 50000,
+
+          // url-loader sets mimetype if it's passed.
+          // Without this it derives it from the file extension
+          mimetype: 'application/font-woff',
+
+          // Output below fonts directory
+          name: './fonts/[name].[ext]'
+        }
+      },
+      {
         test: /\.(svg)$/,
-        loader: 'string-loader',
-      }, {
+        loader: 'string-loader'
+      },
+      {
         test: /\.(png|jpg|gif)$/,
-        loader: 'url-loader',
-      }, {
-        include: [
-          path.resolve(__dirname, '../src/components/variables.scss'),
-          path.resolve(__dirname, '../src/components/mixins.scss'),
-        ],
         loader: 'file-loader',
         options: {
-          name: '[name].[ext]',
-        },
-      }, {
-        include: [
-          path.resolve(__dirname, '../src/components/common.scss'),
-        ],
-        loader: ExtractTextPlugin.extract(['css-loader', 'sass-loader']),
-      },
-    ],
+          name: '[name].[ext]?[hash]'
+        }
+      }
+    ]
   },
   resolveLoader: {
     alias: {
       // necessary to to make lang="scss" work in test when using vue-loader's ?inject option
       // see discussion at https://github.com/vuejs/vue-loader/issues/724
-      'scss-loader': 'sass-loader',
-    },
+      'scss-loader': 'sass-loader'
+    }
   },
   plugins: [
-    new ExtractTextPlugin('common.css'),
+    // Bundle Analyzer
+    new BundleAnalyzerPlugin(),
+    // Define global environment variables
+    new webpack.EnvironmentPlugin(process.env),
+    // Extract CSS
+    new ExtractTextPlugin({
+      filename: 'common.css',
+      disable: false,
+      allChunks: true
+    }),
+    // Hot Module Replacement
+    new webpack.HotModuleReplacementPlugin(),
+    // Common Chunk Splitting
+    new webpack.optimize.CommonsChunkPlugin({
+      // filename: "vendor.js"
+      // (Give the chunk a different name)
+      name: 'vendor',
+      // (with more entries, this ensures that no other module
+      //  goes into the vendor chunk)
+      minChunks: function (module) {
+        // This prevents stylesheet resources with the .css or .scss extension
+        // from being moved from their original chunk to the vendor chunk
+        if (module.resource && (/^.*\.(css|scss)$/).test(module.resource)) {
+          return false
+        }
+        // split the vendor specific modules
+        return module.context && module.context.includes('node_modules')
+      }
+    })
   ],
-};
+  resolve: {
+    alias: {
+      // Global paths
+      '@source': path.resolve(__dirname, '../src'),
+      '@public': path.resolve(__dirname, '../public'),
```

Comment:
Why don't use absolute paths (`/src/...` and `/public`) instead of this aliases?